hepcrawl/parsers/jats.py

```diff
@@ -344,10 +359,12 @@ def get_author_affiliations(self, author_node):
 
         return affiliations
 
-    @staticmethod
-    def get_author_emails(author_node):
+    def get_author_emails(self, author_node):
         """Extract an author's email addresses."""
         emails = author_node.xpath('.//email/text()').extract()
+        referred_ids = author_node.xpath('.//xref[@ref-type="aff"]/@rid').extract()
+        aff_emails = [self.get_emails_from_refs(rid) for rid in referred_ids]
```

Comment:
if you do the change suggested on the next line, you can replace this by a generator comprehension, as `list.extend` can take an iterator:
`aff_emails = (self.get_emails_from_refs(rid) for rid in referred_ids)`